### ISSUE TITLE
testpackage in golangci-lint does not report any error (while standalone it does)

### ISSUE THREAD

**Issue by hlubek** [golangci/golangci-lint#3424]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched for similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

The `testpackage` linter does not report any error, even if I have a test file that doesn't use a `_test` package.
I enabled the linter and tried the default and a custom config in `linters-settings`.

When running `github.com/maratori/testpackage` standalone (v1.1.0)  on the same codebase it does report the expected errors.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.50.1 built from 8926a95 on 2022-10-22T10:48:48Z
```
</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
run:
  concurrency: 4
  timeout: 5m
  tests: false

output:
  format: line-number

linters:
  # Disable all linters, so we enable linters explicitly
  disable-all: true

  enable:
    ### Default linters for golangci-lint
    # Errcheck is a program for checking for unchecked errors in go programs. These unchecked errors can be critical bugs in some cases
    - errcheck
    # Linter for Go source code that specializes in simplifying code
    - gosimple
    # Vet examines Go source code and reports suspicious constructs, such as Printf calls whose arguments do not align with the format string
    - govet
    # Detects when assignments to existing variables are not used
    - ineffassign
    # It's a set of rules from staticcheck
    - staticcheck
    # Like the front-end of a Go compiler, parses and type-checks Go code
    - typecheck
    # Checks Go code for unused constants, variables, functions and types
    - unused

    ### Additional linters
    # Check for pass []any as any in variadic func(...any)
    - asasalint
    # Simple linter to check that your code does not contain non-ASCII identifiers
    - asciicheck
    # Checks for dangerous unicode character sequences
    - bidichk
    # Checks whether HTTP response body is closed successfully
    - bodyclose
    # A linter that detects struct contained context.Context field
    - containedctx
    # Check whether the function uses a non-inherited context
    - contextcheck
    # Disable cyclomatic complexity - it doesn't help much, because of many error handling conditionals
    # - cyclop
    # Disable declaration order style check
    # - decorder
    # Go linter that checks if package imports are in a list of acceptable packages
    - depguard
    # Disable dogsled style check
    # - dogsled
    # Disable duplicate code style check because of false positives
    # - dupl
    # Disable duplicate word checks in comments
    # - dupword
    # Check for two durations multiplied together
    - durationcheck
    # Checks types passed to the json encoding functions
    - errchkjson
    # Checks that sentinel errors are prefixed with the Err and error types are suffixed with the Error.
    - errname
    # A linter for that can be used to find code that will cause problems with the error wrapping scheme introduced in Go 1.13.
    - errorlint
    # Disable execinquery
    # - execinquery
    # Check exhaustiveness of enum switch statements
    - exhaustive
    # Disable exhaustruct since we want to have struct literals with default values
    # - exhaustruct
    # Checks for pointers to enclosing loop variables
    - exportloopref
    # Forbids identifiers (e.g. fmt.Println) from being used
    - forbidigo
    # Finds forced type assertions
    - forcetypeassert
    # Disable funlen because we don't want to limit function length to a fixed length
    # - funlen
    # This analyzer checks for global variables and errors on any found.
    - gochecknoglobals
    # Checks that no init functions are present in Go code
    - gochecknoinits
    # Computes and checks the cognitive complexity of functions
    - gocognit
    # Finds repeated strings that could be replaced by a constant
    - goconst
    # Provides diagnostics that check for bugs, performance and style issues
    - gocritic
    # Disable gocyclo because we already use gocognit
    # - gocyclo
    # Disable godot, because we don't enforce the comments as full sentences internally
    # - godot
    # Disable godox for now, since we have some TODOs still pending
    # - godox
    # Golang linter to check the errors handling expressions
    - goerr113
    # Disable gofmt
    # - gofmt
    # Disable gofumpt
    # - gofumpt
    # Disable goheader
    - goheader
    # Disable goimports
    # - goimports
    # Disable gomnd (magic numbers) for now - too many false positives, but good idea to have
    # - gomnd
    # Disable gomoddirectives
    # - gomoddirectives
    # Disable gomodguard
    # - gomodguard
    # Checks that printf-like functions are named with f at the end
    - goprintffuncname
    # Inspects source code for security problems
    - gosec
    # An analyzer to analyze expression groups
    - grouper
    # Enforces consistent import aliases
    - importas
    # A linter that checks the number of methods inside an interface
    - interfacebloat
    # Disable ireturn
    # - ireturn
    # Disable lll
    # - lll
    # Disable loggercheck
    # - loggercheck
    # Measures the maintainability index of each function
    - maintidx
    # Finds slice declarations with non-zero initial length
    - makezero
    # Finds commonly misspelled English words in comments
    - misspell
    # Finds naked returns in functions greater than a specified function length
    - nakedret
    # Disable nestif
    # - nestif
    # Finds code that returns nil even if it checks that the error is not nil
    - nilerr
    # Disable nilnil
    # - nilnil
    # Disable nlreturn
    # - nlreturn
    # Finds sending http request without context.Context
    - noctx
    # Reports ill-formed or insufficient nolint directives
    - nolintlint
    # Disable nonamedreturns
    # - nonamedreturns
    # Checks for misuse of Sprintf to construct a host with port in a URL
    - nosprintfhostport
    # Disable paralleltest
    # - paralleltest
    # Finds slice declarations that could potentially be pre-allocated
    - prealloc
    # Find code that shadows one of Go's predeclared identifiers
    - predeclared
    # Disable promlinter
    # - promlinter
    # Checks that package variables are not reassigned
    - reassign
    # Fast, configurable, extensible, flexible, and beautiful linter for Go. Drop-in replacement of golint.
    - revive
    # Disable rowserrcheck because of generics
    # - rowserrcheck
    # Disable sqlclosecheck because of generics
    # - sqlclosecheck
    # Disable stylecheck because it duplicates checks of revive
    # - stylecheck
    # Checks the naming convention of fields in struct tags
    - tagliatelle
    # Analyzer that detects using os.Setenv instead of t.Setenv
    - tenv
    # Disable testableexamples because we have none
    # - testableexamples
    # Check tests are defined in a separate test package (_test) # TODO This is currently broken in golangci-lint
    - testpackage

linters-settings:
  errcheck:
    check-type-assertions: true
    check-blank: true
  govet:
    check-shadowing: false
    settings:
      printf:
        funcs:
          - github.com/apex/log.Debugf
          - github.com/apex/log.Infof
          - github.com/apex/log.Warnf
          - github.com/apex/log.Errorf
          - github.com/apex/log.Fatalf
          - github.com/apex/log.Fatalf
          - (github.com/apex/log.Interface).Debugf
          - (github.com/apex/log.Interface).Infof
          - (github.com/apex/log.Interface).Warnf
          - (github.com/apex/log.Interface).Errorf
          - (github.com/apex/log.Interface).Fatalf
  depguard:
    include-go-root: true
    packages-with-error-message:
      - github.com/pkg/errors: 'use github.com/friensofgo/errors instead'
      - github.com/go-errors/errors: 'use github.com/friensofgo/errors instead'
      - github.com/pingcap/errors: 'use github.com/friensofgo/errors instead'
      - log: 'use github.com/apex/log instead'
  exhaustive:
    default-signifies-exhaustive: true

issues:
  exclude-rules:
    # Allow init functions for migrations
    - path: persistence/migrations
      linters:
        - gochecknoinits
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.19.2 darwin/arm64
GO111MODULE="on"
GOARCH="arm64"
GOBIN=""
GOCACHE=".../Library/Caches/go-build"
GOENV=".../Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="arm64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE=".../go/pkg/mod"
GONOPROXY="github.com/exadot"
GONOSUMDB="github.com/exadot"
GOOS="darwin"
GOPATH=".../go"
GOPRIVATE="github.com/exadot"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_arm64"
GOVCS=""
GOVERSION="go1.19.2"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=".../my-project/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/03/6gs32_tx5px6lffj_v58j0m80000gp/T/go-build4242954114=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /Users/hlubek/Dev/Work/broodi/trainee-portal/backend /Users/hlubek/Dev/Work/broodi/trainee-portal /Users/hlubek/Dev/Work/broodi /Users/hlubek/Dev/Work /Users/hlubek/Dev /Users/hlubek /Users /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 49 linters: [asasalint asciicheck bidichk bodyclose containedctx contextcheck depguard durationcheck errcheck errchkjson errname errorlint exhaustive exportloopref forbidigo forcetypeassert gochecknoglobals gochecknoinits gocognit goconst gocritic goerr113 goheader goprintffuncname gosec gosimple govet grouper importas ineffassign interfacebloat maintidx makezero misspell nakedret nilerr noctx nolintlint nosprintfhostport prealloc predeclared reassign revive staticcheck tagliatelle tenv testpackage typecheck unused] 
INFO [loader] Go packages loading at mode 575 (compiled_files|deps|files|name|exports_file|imports|types_sizes) took 380.482084ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 35.21775ms 
INFO [linters_context] importas settings found, but no aliases listed. List aliases under alias: key. 
INFO [linters_context/goanalysis] analyzers took 52.774225718s with top 10 stages: buildir: 11.967687494s, buildssa: 9.287250801s, bidichk: 2.071219503s, the_only_name: 2.054733362s, nilness: 1.759365458s, gocritic: 1.305203754s, exhaustive: 1.270033567s, fact_deprecated: 1.267305173s, gosec: 1.257232415s, inspect: 1.238302582s 
INFO [runner] Issues before processing: 3185, after processing: 0 
INFO [runner] Processors filtering stat (out/in): cgo: 3185/3185, skip_files: 3185/3185, identifier_marker: 1203/1203, filename_unadjuster: 3185/3185, path_prettifier: 3185/3185, skip_dirs: 3185/3185, autogenerated_exclude: 1203/3185, exclude: 1203/1203, exclude-rules: 151/1203, nolint: 0/151 
INFO [runner] processing took 58.798541ms with stages: exclude-rules: 26.393083ms, identifier_marker: 12.953334ms, path_prettifier: 7.530125ms, autogenerated_exclude: 5.647042ms, nolint: 5.42725ms, skip_dirs: 565.959µs, cgo: 181.833µs, filename_unadjuster: 96.791µs, max_same_issues: 1.041µs, uniq_by_line: 376ns, source_code: 333ns, sort_results: 250ns, skip_files: 209ns, severity-rules: 208ns, exclude: 207ns, path_shortener: 125ns, diff: 124ns, max_from_linter: 84ns, path_prefixer: 84ns, max_per_file_from_linter: 83ns 
INFO [runner] linters took 8.115013292s with stages: goanalysis_metalinter: 8.056138125s 
INFO File cache stats: 364 entries of total size 1.9MiB 
INFO Memory: 87 samples, avg is 726.1MB, max is 1102.9MB 
INFO Execution took 8.538507584s                  
```

</details>

### Code example or link to a public repository

Projet is not public

**Comment by ldez** [comment]
Hello,

can you provide a minimal reproducible example?

**Comment by maratori** [comment]
It may happen because of your configuration. Linter ignores test files because `run.tests` is `false`.

**Comment by hlubek** [comment]
@maratori Indeed! Thanks for the hint and fast reaction. I didn't look at this setting. 🤦‍♂️
